### PR TITLE
Minor additions to writing for site docs

### PR DESCRIPTION
**Pull request by parlough**
This isn't extensive, but I wanted to make some update to the writing docs here as well as add some basic information on new topics, such as when to use `{% raw %}`.

I hope to eventually write more proper guidelines from scratch, but this covers some important topics for now.

### CODE REVIEW

File: doc/writing-for-dart-and-flutter-websites.md [hunk 1]
```diff
@@ -168,6 +204,7 @@ js:
 
 The **title:** tag is used for the title displayed on the page 
 and (if no **short-title** tag is present) in the tab.
+Titles should use sentence case.
 
 The **description:** tag replaces the default sharing text
 (what you see on Twitter, Facebook, G+).
```
Comment (sfshaza2) [inline doc/writing-for-dart-and-flutter-websites.md:210]: Maybe also mention that the description text can't have line breaks?

File: doc/markdown.md [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
-        class C { ... }
-        ```
+    ````markdown
+    ```dart
+    class C { ... }
+    ```
+    ````
 
  1. Use the [prettify][] Jekyll block tag, supported via a custom plugin
     designed for use on our sites:
```
Comment (sfshaza2) [inline doc/markdown.md:20]: I thought we no longer use or recommend prettify?

File: doc/markdown.md [hunk 3]
```diff
@@ -50,43 +51,62 @@ conventions make it easier to work with lists.
...
 
-    Now that we have a `TabController` to work with, we can create our
-    tabs using the [TabBar](https://docs.flutter.io/flutter/material/TabController-class.html) widget.
-    In this example, we'll create a `TabBar` with 3 [Tab](https://docs.flutter.io/flutter/material/Tab-class.html)
-    widgets and place it within an [AppBar](https://docs.flutter.io/flutter/material/AppBar-class.html).
+```markdown
+Now that we have a `TabController` to work with, we can create our
+tabs using the [TabBar](https://api.flutter.dev/flutter/material/TabBar-class.html) widget.
+In this example, we'll create a `TabBar` with 3 [Tab](https://api.flutter.dev/flutter/material/Tab-class.html)
+widgets and place it within an [AppBar](https://api.flutter.dev/flutter/material/AppBar-class.html).
```
Comment (sfshaza2) [inline doc/markdown.md:83]: Hmmmm. This is actually a bad doc example, as it should use second person, not first: https://developers.google.com/style/person

File: doc/writing-for-dart-and-flutter-websites.md [hunk 4]
```diff
@@ -123,24 +113,70 @@ Keep code valid and foolproof. People love to copy-paste code.
...
+[Liquid template engine][liquid]
+for processing templates.
+This support is responsible for variable support such as ``{{site.dart-api}}``,
+[includes][], custom-built tags, and [filters][].
+
+You can use this support to generate content, include files,
+avoid repetition, and more.
+To learn more, see the [Liquid documentation][liquid].
+
+If you need to use Liquid syntax in text or a code snippet,
```
Comment (sfshaza2) [inline doc/writing-for-dart-and-flutter-websites.md:152]: Thanks for adding this!

**Review by sfshaza2 — COMMENTED** [review]
Thanks for doing this, @parlough!

File: doc/markdown.md (hunk 3 above)
Comment (parlough) [reply doc/markdown.md]: Nice catch, I didn't even read it while updating the links. Thanks!

File: doc/markdown.md (hunk 2 above)
Comment (parlough) [reply doc/markdown.md]: You still need to use it if you want highlighting on a snippet that does not have a code-excerpt. I've added a note about that until we can migrate completely away from it. Related issue: https://github.com/dart-lang/site-www/issues/3846

**Review by sfshaza2 — APPROVED** [review]
lgtm. Thanks, @parlough!

**Pull request merged**